directembedding/src/main/scala/ch/epfl/directembedding/Utils.scala

```diff
@@ -10,11 +12,31 @@ trait MacroModule {
 }
 
 trait DirectEmbeddingModule extends MacroModule {
-  import c.universe._
-  def lift(tree: Tree): Tree
+  val dslName: String
+  val dslEndpointMethod: String
+  val virtualizationConfig: c.universe.Type
+  val liftMethodName: String
+  val logLevel: Int
 }
 
-trait DirectEmbeddingUtils {
-  def debugLevel: Int
-  def log(s: => String, level: Int = 0) = if (debugLevel < level) println(s)
+trait DslVirtualizationConfig
+
+trait DirectEmbeddingUtils extends DirectEmbeddingModule with TransformationUtils {
+  import c.universe._
+
+  def logTree(t: Tree, level: Int = 0) = {
+    log(s"$t", level)
+    log(showRaw(t, printTypes = true), level + 1)
+  }
+
+  def logIndented[T](e: T, indent: Int, level: Int = 0) = if (debugLevel > level) {
+    print(" " * indent)
+    super.log(s"$e", level)
+  }
+
+  // A hack to make the DSLVirtualization import statement 'typesafe'
+  def importType(tpe: c.universe.Type): Tree = {
```

Comment:
Haven't we gone back in a circle then, since the resulting type will be a `String` anyways?

By accepting a `Type`, we force the author to use the reflection api as in [here](https://github.com/directembedding/directembedding/pull/23/files#diff-d1429d3c359afa93eb64775fe183f5c8R39)